### DISCUSSION TITLE
SaltRenderError: Jinja variable the template 'salt/map.jinja' (imported on line 1) .......

### DISCUSSION THREAD

**Discussion by DebianGuru** [Security-Onion-Solutions/securityonion#12610] [2.4]
### Version

2.4.50

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

12

### RAM

20gb

### Storage for /

200g

### Storage for /nsm

1.5TB

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

Salt never comes up in /opt/so/logs/salt/minion. I see; [salt.state       :4325][CRITICAL][75409] Rendering SLS 'base:salt.minion' failed: Jinja variable the template 'salt/map.jinja' (imported on line 1) does not export the requested name 'UPGRADECOMMAND'; line 34

/opt/so/saltstack/defaults/salt/salt/map.jinga contains:
 

> {% import_yaml 'salt/minion.defaults.yaml' as saltminion %}
> {% set SALTVERSION = saltminion.salt.minion.version %}
> 
> {% if grains.os == 'Ubuntu' %}
>   {% set SPLITCHAR = '+' %}
>   {% set SALTNOTHELD = salt['cmd.run']('apt-mark showhold | grep -q salt ; echo $?', python_shell=True) %}
>   {% set SALTPACKAGES = ['salt-common', 'salt-master', 'salt-minion'] %}
>   {% set SALT_STATE_CODE_PATH = '/usr/lib/python3/dist-packages/salt/states' %}
>   {% set SALT_MODULE_CODE_PATH = '/usr/lib/python3/dist-packages/salt/modules' %}
>   {% set PYTHONINFLUXVERSION = '5.3.1' %}
>   {% set PYTHON3INFLUX= 'influxdb == ' ~ PYTHONINFLUXVERSION  %}
>   {% set PYTHON3INFLUXDEPS= ['certifi', 'chardet', 'python-dateutil', 'pytz', 'requests'] %}
>   {% set PYTHONINSTALLER = 'pip' %}
>   {% set SYSTEMD_UNIT_FILE = '/lib/systemd/system/salt-minion.service' %}
> {% else %}
>   {% set SPLITCHAR = '-' %}
>   {% set SALTNOTHELD = salt['cmd.run']('yum versionlock list | grep -q salt ; echo $?', python_shell=True) %}
>   {% set SALTPACKAGES = ['salt', 'salt-master', 'salt-minion'] %}
>   {% set SALT_STATE_CODE_PATH = '/usr/lib/python3.6/site-packages/salt/states' %}
>   {% set SALT_MODULE_CODE_PATH = '/usr/lib/python3.6/site-packages/salt/modules' %}
>   {% set PYTHONINFLUXVERSION = '5.3.1' %}
>   {% set PYTHON3INFLUX= 'securityonion-python3-influxdb' %}
>   {% set PYTHON3INFLUXDEPS= ['python36-certifi', 'python36-chardet', 'python36-dateutil', 'python36-pytz', 'python36-requests'] %}
>   {% set PYTHONINSTALLER = 'pkg' %}
>   {% set SYSTEMD_UNIT_FILE = '/usr/lib/systemd/system/salt-minion.service' %}
> {% endif %}
> 
> {% set INSTALLEDSALTVERSION = salt['pkg.version']('salt-minion').split(SPLITCHAR)[0] %}
> 
> {% if grains.saltversion|string != SALTVERSION|string %}
>   {% if grains.os|lower in ['centos', 'redhat'] %}
>       {% set UPGRADECOMMAND = 'yum clean all ; /usr/sbin/bootstrap-salt.sh -s 120 -r -F -x python3 stable ' ~ SALTVERSION %}
>   {% elif grains.os|lower == 'ubuntu' %}
>     {% set UPGRADECOMMAND = '/usr/sbin/bootstrap-salt.sh -s 120 -r -F -x python3 stable ' ~ SALTVERSION %}
>   {% endif %}
> {% else %}
>   {% set UPGRADECOMMAND = 'echo Already running Salt Minion version ' ~ SALTVERSION %}
> {% endif %}

**Note line 34 is**
>     {% set UPGRADECOMMAND = '/usr/sbin/bootstrap-salt.sh -s 120 -r -F -x python3 stable ' ~ SALTVERSION %}

/opt/so/saltstack/default/salt/salt/minion.defaults.yaml  contains:

> #version cannot be used elsewhere in this pillar as soup is grepping for it to determine if Salt needs to be patched
> # When updating the salt version, also update the version in securityonion-builds/images/iso-task/Dockerfile and saltify function in so-functions
> salt:
>   minion:
>     version: 3004.2
>     check_threshold: 3600 # in seconds, threshold used for so-salt-minion-check. any value less than 600 seconds may cause a lot of salt-minion restarts since the job to touch the file occurs every 5-8 minutes by default
>     service_start_delay: 30 # in seconds.
> 

 /opt/so/saltstack/default/salt/salt/minion.sls  begins with: 

> {% from 'salt/map.jinja' import UPGRADECOMMAND with context %}
> {% from 'salt/map.jinja' import SALTVERSION %}
> {% from 'salt/map.jinja' import INSTALLEDSALTVERSION %}
> {% from 'salt/map.jinja' import SALTNOTHELD %}
> {% from 'salt/map.jinja' import SALTPACKAGES %}
> {% from 'salt/map.jinja' import SYSTEMD_UNIT_FILE %}
> {% import_yaml 'salt/minion.defaults.yaml' as SALTMINION %}
> {% set service_start_delay = SALTMINION.salt.minion.service_start_delay %}

> .
> .
> .

If I understand the log, salt is puking on the first line on the minion.sls file where it's trying to import the value for UPGRADECOMMAND which IS defined up in the map.jinja file.  Is the system claiming that the value of UPGRADECOMMAND in the map.jinja file not being exported so it can be referenced by the minion.sls file.

This all broke after I ran `so-elastic-fleet-reset` .

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
What does `dnf info salt*` show?

**Reply by DebianGuru** [reply]
@cm-ops ,
I really appreciate the help with this, unfortunately, due to time constraints, I ended up re-running the installer from scratch.  Things are working now.  I really wish I could have found the root cause, but I was under pressure to get things running again.  Thanks again for all of your help.
